lib/apiScenario/apiScenarioRunner.ts

```diff
@@ -194,23 +197,77 @@ export class ApiScenarioRunner {
         case "armTemplateDeployment":
           await this.executeArmTemplateStep(step, stepEnv, scope);
           break;
+        case "armRoleAssignment":
+          await this.executeArmRoleAssignmentStep(step, stepEnv);
+          break;
       }
     } catch (e) {
       throw new Error(`Failed to execute step ${step.step}: ${e.message} \n${e.stack}`);
     }
   }
 
+  private async executeArmRoleAssignmentStep(step: StepRoleAssignment, env: VariableEnv) {
+    const parameters = {
+      scope: step.roleAssignment.scope,
+      roleAssignmentName: "{{$guid}}",
+      "api-version": "2022-04-01",
+    };
+
+    const roleId =
+      step.roleAssignment.roleDefinitionId ??
+      AzureBuiltInRoles.find((r) => r.roleName === step.roleAssignment.roleName)?.roleDefinitionId;
+
+    if (roleId === undefined) {
+      throw new Error(
+        `Cannot find role definition id for role name ${step.roleAssignment.roleName}`
+      );
+    }
+
+    const req: ApiScenarioClientRequest = {
+      host: DEFAULT_ARM_ENDPOINT,
```

Comment:
host should be an option

Reply:
it's required in ApiScenarioClientRequest